### PR TITLE
Feat: removing flared vented ng

### PR DESCRIPTION
**Pull request by pbastia**
Removing 'Flared Natural Gas' and 'Vented Natural Gas' as options for fuel.
- [x] update the `FuelField`  to render an alert if the fuel is archived
- [x] update the `FuelRowIdField` to still render the fuel name when the fuel is deprecated
- [x] unit tests for FuelField/FuelRowIdField
- [x] updates to e2e tests if necessary

### CODE REVIEW

File: schema/data/prod/fuel.sql [hunk 1]
```diff
@@ -92,9 +92,9 @@ values
   -- Flared Natural Gas was previously named Flared Natural Gas CH4
   -- It was renamed and the swrs_fuel_mapping_id was updated as applicants for the CIIP 2019 applications
   -- were instructed to report an amount of natural gas under this fuel
-  (86,'Flared Natural Gas','m3', 'active', 95),
+  (86,'Flared Natural Gas','m3', 'archived', 95),
```
Comment (matthieu-foucault) [inline schema/data/prod/fuel.sql:95]: I am thinking that we should add a `comments` column to the `fuel` table, to describe why those fuels are archived (and that natural gas should be used instead). The comment above for Flared Natural Gas CH4 would also go in that column.
This wouldn't be intended to be displayed, but as a metadata so that anyone looking at this table in the database can see what's up with those archived fuels.

File: app/containers/Forms/FuelField.tsx [hunk 2]
```diff
@@ -26,17 +33,32 @@ const FuelField: React.FunctionComponent<Props> = (props) => {
...
   };
 
-  return <ObjectField {...props} onChange={handleChange} />;
+  return (
+    <>
+      {isFuelArchived && (
+        <Alert variant="danger">
+          <strong>Warning:</strong> This fuel has been archived. Please remove
+          it and select an appropriate alternative.
+        </Alert>
```
Comment (matthieu-foucault) [inline app/containers/Forms/FuelField.tsx:42]: Will this alert be displayed to the reviewer when looking at an existing application? If so it might be confusing when they look at 2019 applications?

File: app/containers/Forms/ProductField.tsx [hunk 3]
```diff
@@ -27,14 +27,13 @@ export const ProductFieldComponent: React.FunctionComponent<Props> = (
...
 
-  // TODO: Clean up this function with array.some() as noted in https://github.com/bcgov/cas-ciip-portal/pull/621
-  const productIsPublished = (formData, query) => {
-    const product = query.allProducts.edges.find(
-      ({node}) => node.rowId === formData.productRowId
-    )?.node;
-    if (product?.productState === 'PUBLISHED' || !product) return true;
-    return false;
-  };
+  const productIsPublished =
```
Comment (dleard) [inline app/containers/Forms/ProductField.tsx:30]: Can we undo these changes? This file is going through major changes in #1580 & this is going to cause conflicts

File: app/containers/Forms/ProductRowIdField.tsx [hunk 4]
```diff
@@ -76,15 +76,11 @@ export const ProductRowIdFieldComponent: React.FunctionComponent<Props> = (
...
-    <div>
-      <InputGroup className="mb-3">
-        <InputGroup.Prepend>
-          <InputGroup.Text>
-            {allProducts.archivedProductNames[archivedIndex]}
-          </InputGroup.Text>
-        </InputGroup.Prepend>
-      </InputGroup>
-    </div>
+    <InputGroup>
```
Comment (dleard) [inline app/containers/Forms/ProductRowIdField.tsx:79]: ditto undo changes as this file also changeds a lot in #1580

File: app/containers/Forms/FuelField.tsx (hunk 2 above)
Comment (pbastia) [reply app/containers/Forms/FuelField.tsx:42]: Yes, they show up on the summary page as well, I replicated the behaviour and look that was used for archived products. It could be a low impact improvement to clean this up, maybe a tech debt card?

File: schema/verify/tables/fuel_001.sql [hunk 5]
```diff
@@ -0,0 +1,11 @@
+-- Verify ggircs-portal:tables/fuel_001 on pg
+
+BEGIN;
```
Comment (dleard) [inline schema/verify/tables/fuel_001.sql:3]: nitpicking here...toLowerCase()

File: app/containers/Forms/FuelRowIdField.tsx [hunk 6]
```diff
@@ -17,26 +18,63 @@ export const FuelRowIdFieldComponent: React.FunctionComponent<Props> = (
...
+      </InputGroup.Text>
+    </InputGroup>
+  );
 };
 
 export default createFragmentContainer(FuelRowIdFieldComponent, {
   query: graphql`
     fragment FuelRowIdField_query on Query {
-      allFuels(condition: {state: "active"}) {
+      activeFuels: allFuels(condition: {state: "active"}) {
```
Comment (dleard) [inline app/containers/Forms/FuelRowIdField.tsx:69]: going forward do we want to use `filter` rather than `condition` for consistency? (This is a question, not necessarily a requested change). I've been replacing condition with filter when refactoring things.
Also: 
```suggestion
      activeFuels: allFuels(condition: {state: "active"} orderBy: NAME_ASC) {
```

**Review by dleard — CHANGES_REQUESTED** [review]

File: app/containers/Forms/FuelField.tsx (hunk 2 above)
Comment (matthieu-foucault) [reply app/containers/Forms/FuelField.tsx:42]: It can be cleaned up later, but it's not tech tebt, it's a feature

File: app/containers/Forms/FuelRowIdField.tsx (hunk 6 above)
Comment (pbastia) [reply app/containers/Forms/FuelRowIdField.tsx:69]: Happy to indulge! I doubt there is any performance concern between using the postgraphile filter plugin and the default graphQL syntax

**Review by dleard — APPROVED** [review]
Awesome! :shipit:

**Review by matthieu-foucault — APPROVED** [review]
